src/reports.cpp

```diff
@@ -874,7 +874,7 @@ static int attack_info(reports::context & rc, const attack_type &at, config &res
 		}
 
 		typedef std::pair<int, std::set<std::string>> resist_units;
-		for (const resist_units &resist : resistances) {
+		for (const resist_units resist : resistances) {
```

Comment:
resist_units is a quote nontrivial type to copy, you shouldn't copy it here, if the old code copies it we should figure out why that is instead of removing the `&`

Reply:
actualyl here is why i does a copy: you need `<const int, ` instead of `<int, ` in the line above, please add that const and add the `&` back.

Reply:
Updated as well.

And for reference, the full text of the warning was:
```
src/reports.cpp:877:28: error: loop variable 'resist' has type 'const resist_units &' (aka 'const pair<int, set<basic_string<char> > > &') but is initialized with type 'std::pair<const int, std::set<std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::__cxx11::basic_string<char> > > >' resulting in a copy [-Werror,-Wrange-loop-construct]
                for (const resist_units &resist : resistances) {
                                         ^
src/reports.cpp:877:8: note: use non-reference type 'resist_units' (aka 'pair<int, set<basic_string<char> > >') to keep the copy or type 'const std::pair<const int, std::set<std::__cxx11::basic_string<char>, std::less<std::__cxx11::basic_string<char> >, std::allocator<std::__cxx11::basic_string<char> > > > &' to prevent copying
                for (const resist_units &resist : resistances) {
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
So it was previously being copied.